pkg/utils/authcheck/authtype.go

```diff
@@ -114,7 +125,12 @@ func getAuthTypeForWorkloadIdentity(ctx context.Context, serviceAccountLister co
 				args.ServiceAccountName)
 		}
 		return "", fmt.Errorf("error getting Kubernetes Service Account: %w", err)
-	} else if kServiceAccount.Annotations[resources.WorkloadIdentityKey] != "" {
+	} else if email := kServiceAccount.Annotations[resources.WorkloadIdentityKey]; email != "" {
+		// Check if email is a valid google service account email.
+		if match := emailRegexp.FindStringSubmatch(email); len(match) == 0 {
```

Comment:
Do we ever expect multiple emails in this? My guess is that we don't need Submatch.

```suggestion
		if match := emailRegexp.FindString(email); match == "" {
```